src/btree/col_srch.c

```diff
@@ -286,11 +286,37 @@ __wt_col_search(
      * because column-store files are dense, but in this case the caller searched past the end of
      * the table.
      */
-    cbt->ins_head = WT_COL_APPEND(page);
-    if ((cbt->ins = __col_insert_search(cbt->ins_head, cbt->ins_stack, cbt->next_stack, recno)) ==
-      NULL)
+    ins_head = WT_COL_APPEND(page);
+    ins = __col_insert_search(ins_head, cbt->ins_stack, cbt->next_stack, recno);
+    if (ins == NULL) {
+        /*
+         * Set this, otherwise the code in cursor_valid will assume there's no on-disk value
+         * underneath ins_head.
+         */
+        if (page->type != WT_PAGE_COL_FIX)
```

Comment:
I don't think we should set it here. We should set it when we goto this section. It's possible we end up here without finding anything on the disk image as well.